executor/runcexecutor/executor.go

```diff
@@ -187,6 +192,9 @@ func (w *runcExecutor) Exec(ctx context.Context, meta executor.Meta, root cache.
 	if meta.ReadonlyRootFS {
 		opts = append(opts, containerdoci.WithRootFSReadonly())
 	}
+	if w.cgroupParent != "" {
+		ctx = namespaces.WithNamespace(ctx, w.cgroupParent)
```

Comment:
Probably we should explicitly specify WithNamespacedCgroup() 
https://github.com/containerd/containerd/blob/3f42445e38d1081f4b8c3b8d7d1ed1860198ed7a/oci/spec_opts_unix.go#L256

Reply:
@AkihiroSuda I think the functionality of `WithNamespacedCgroup()` is also achieved by https://github.com/moby/buildkit/blob/49906c62925ed429ec9174a0b6869982967f1a39/executor/oci/spec_unix.go#L39 which also populates `CgroupPath` field with `filepath.Join("/", ns, id)`. (at very end of the function call)
What is really missing here is the namespace that needs to be set into the context so that the spec generation can use it in below https://github.com/moby/buildkit/blob/49906c62925ed429ec9174a0b6869982967f1a39/vendor/github.com/containerd/containerd/oci/spec_unix.go#L79

Reply:
Yes, it is covered by the existing code, but I feel explicitly setting the opt would be better for readability and  maintainability.

Reply:
@AkihiroSuda Do you mean something like this? 
```
	if w.cgroupParent != "" {
		ctx = namespaces.WithNamespace(ctx, w.cgroupParent)
		opts = append(opts, containerdoci.WithNamespacedCgroup())
	}
```
I see your point to make it more explicit on the top level, although I still think the existing code with proper comments can lead to the same way. Especially with the `WithNamespacedCgroup ` SpecOpts added, the same block of code will run twice.